### PR TITLE
test: add mock consensus to end to end test for batcher

### PR DESCRIPTION
**Pull request by ArniStarkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1296)

### CODE REVIEW

**Comment by ArniStarkware** [comment]
* **#1296** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1296?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ArniStarkware and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by giladchase — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1296)* status: all files reviewed, 3 unresolved discussions (waiting on @ArniStarkware and @yair-starkware)

___
*[`crates/tests-integration/src/integration_test_setup.rs` line 107 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8q3sC267K1QdktPqze:-O8q3sC267K1QdktPqzf:bk7ki7e) ([raw file](https://github.com/starkware-libs/sequencer/blob/fa759c61f17ab0c574c25b2e91c4440763963e4f/crates/tests-integration/src/integration_test_setup.rs#L107)):*
> ```XML
>     }
> 
>     // TODO(Arni): delete function and mempool_client.
> ```

I'm not sure if we neccesarily want to get rid of this behavior.
For the e2e integration test we wont, for sure, but for other types of flow/integraiton tests we might want to trigger stuff manually.
Not sure, but I suggest at least to change the wording to reasses once we have more tests in mind.



_Code quote:_
```XML
    // TODO(Arni): delete function and mempool_client.
```
___
*[`crates/tests-integration/src/mock_consensus_manager.rs` line 36 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8q3Eva8n3te2Dw_5J4:-O8q3Eva8n3te2Dw_5J5:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/fa759c61f17ab0c574c25b2e91c4440763963e4f/crates/tests-integration/src/mock_consensus_manager.rs#L36)):*
> ```XML
> 
>     #[allow(dead_code)]
>     async fn decition_reached(&self, input: DecisionReachedInput) {
> ```





_Suggestion:_
```XML
    async fn decision_reached(&self, input: DecisionReachedInput) {
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 45 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8q4TFa16BeID0Nw_2F:-O8q4TFa16BeID0Nw_2G:b1jhb6m) ([raw file](https://github.com/starkware-libs/sequencer/blob/fa759c61f17ab0c574c25b2e91c4440763963e4f/crates/tests-integration/tests/end_to_end_test.rs#L45)):*
> ```XML
>     let mempool_txs = mock_running_system.get_txs(4).await;
> 
>     mock_running_system.mock_consensus_manager.run_consensus_for_end_to_end_test(height).await;
> ```

Some questions about this.

Should we initialize the batcher with this value for now?
If not, and we're creating this API, can you give examples of flows that will use different values?



_Code quote:_
```XML
    let height = BlockNumber(1);
    // Test.
    let mempool_txs = mock_running_system.get_txs(4).await;

    mock_running_system.mock_consensus_manager.run_consensus_for_end_to_end_test(height).await;
```

**Review by ArniStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1296)* status: 1 of 6 files reviewed, 1 unresolved discussion (waiting on @giladchase and @yair-starkware)

___
*[`crates/tests-integration/src/integration_test_setup.rs` line 107 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8q3sC267K1QdktPqze:-O8qAAW4EQEVFY8ZvqpS:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/fa759c61f17ab0c574c25b2e91c4440763963e4f/crates/tests-integration/src/integration_test_setup.rs#L107)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

I'm not sure if we neccesarily want to get rid of this behavior.
For the e2e integration test we wont, for sure, but for other types of flow/integraiton tests we might want to trigger stuff manually.
Not sure, but I suggest at least to change the wording to reasses once we have more tests in mind.
</blockquote></details>

Done.
___
*[`crates/tests-integration/src/mock_consensus_manager.rs` line 36 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8q3Eva8n3te2Dw_5J4:-O8q9ZiK0ZXG72Ssg_Rf:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/fa759c61f17ab0c574c25b2e91c4440763963e4f/crates/tests-integration/src/mock_consensus_manager.rs#L36)):*
> ```XML
> 
>     #[allow(dead_code)]
>     async fn decition_reached(&self, input: DecisionReachedInput) {
> ```

Done.
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 45 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8q4TFa16BeID0Nw_2F:-O8qADvI1S3X0qhMpmVV:bda9fsy) ([raw file](https://github.com/starkware-libs/sequencer/blob/fa759c61f17ab0c574c25b2e91c4440763963e4f/crates/tests-integration/tests/end_to_end_test.rs#L45)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

Some questions about this.

Should we initialize the batcher with this value for now?
If not, and we're creating this API, can you give examples of flows that will use different values?
</blockquote></details>

Let us restrict the start height for now. We can change this to a parameter later if needed.

**Review by yair-starkware — CHANGES_REQUESTED** [review]
> Reviewed 2 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1296)* status: 3 of 6 files reviewed, 2 unresolved discussions (waiting on @ArniStarkware and @giladchase)

*[a discussion](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8rhVXJ4ygero_CasRR:-O8rhVXJ4ygero_CasRS:b-b1fv1p) (no related file):*
Why do we need the mock struct?

___

**Comment by ArniStarkware** [comment]
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

Why do we need the mock struct?
</blockquote></details>

The motivation is to enclose all the temporary code in one place. We do not want to fill the file `end_to_end_test.rs` with garbage.
The mock consensus manager will be deleted once the proper consensus manager is added.

This answers why there is a new file: `crates/tests-integration/src/mock_consensus_manager.rs`. I have no good answer about why it is a struct - other than for consistency (And I am thinking with python).

**Review by giladchase — COMMENTED** [review]
> Reviewed 2 of 5 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1296)* status: all files reviewed, 1 unresolved discussion (waiting on @yair-starkware)

___
*[`crates/tests-integration/src/mock_consensus_manager.rs` line 43 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O93wjf9163BzsQeS9fX:-O93wjf9163BzsQeS9fY:b-byfmbz) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65d9654fa7d5031f4efa5b15620428ec1d829de/crates/tests-integration/src/mock_consensus_manager.rs#L43)):*
> ```XML
>     /// This function should mirror
>     /// [`run_consensus`](papyrus_consensus::manager::run_consensus). It makes requests
>     /// from the batcher client and asserts the expected responses were received.
> ```

Just curious, how does it differ from `run_consensus`?



_Code quote:_
```XML
    /// This function should mirror
    /// [`run_consensus`](papyrus_consensus::manager::run_consensus). It makes requests
    /// from the batcher client and asserts the expected responses were received.
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 45 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8q4TFa16BeID0Nw_2F:-O93vCJAEXN7QFPAk-ry:b-c199z9) ([raw file](https://github.com/starkware-libs/sequencer/blob/fa759c61f17ab0c574c25b2e91c4440763963e4f/crates/tests-integration/tests/end_to_end_test.rs#L45)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

Let us restrict the start height for now. We can change this to a parameter later if needed.
</blockquote></details>

Cool thanks!
I'm still not sure about having to initialize the consensus inside the test though, can this be done in the constructor of mock_running_system?
In other words, will there be an integration where this function is not called? If there isn't I think it should not be part of the test, but included inside the initialization.

**Review by yair-starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1296)* status: all files reviewed, 1 unresolved discussion (waiting on @ArniStarkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8rhVXJ4ygero_CasRR:-O94431d7TcCjDbP5GI1:b3vgm0j) (no related file):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

The motivation is to enclose all the temporary code in one place. We do not want to fill the file `end_to_end_test.rs` with garbage.
The mock consensus manager will be deleted once the proper consensus manager is added.

This answers why there is a new file: `crates/tests-integration/src/mock_consensus_manager.rs`. I have no good answer about why it is a struct - other than for consistency (And I am thinking with python).
</blockquote></details>

IMO you remove the struct and leave just `run_consensus_for_end_to_end_test` that will get the batcher_client from outside

___

**Review by ArniStarkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1296)* status: 4 of 6 files reviewed, 1 unresolved discussion (waiting on @giladchase)

___
*[`crates/tests-integration/src/mock_consensus_manager.rs` line 43 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O93wjf9163BzsQeS9fX:-O94-xio8W98PJQuHoXk:b1tny03) ([raw file](https://github.com/starkware-libs/sequencer/blob/d65d9654fa7d5031f4efa5b15620428ec1d829de/crates/tests-integration/src/mock_consensus_manager.rs#L43)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

Just curious, how does it differ from `run_consensus`?
</blockquote></details>

`run_consensus` is the function run by main of the service consensus manager. It runs indefinitely (there is a [loop]( https://github.com/starkware-libs/sequencer/blob/e96a5c75bdece74be7c1e2a466f3c84c2704b332/crates/sequencing/papyrus_consensus/src/manager.rs#L63) in it). 

This is just a mock—it should not run indefinitely—but it calls the butcher's functions for one round.
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 45 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1296#-O8q4TFa16BeID0Nw_2F:-O93xw-EAYgToyAwSs89:bg03c5s) ([raw file](https://github.com/starkware-libs/sequencer/blob/fa759c61f17ab0c574c25b2e91c4440763963e4f/crates/tests-integration/tests/end_to_end_test.rs#L45)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

Cool thanks!
I'm still not sure about having to initialize the consensus inside the test though, can this be done in the constructor of mock_running_system?
In other words, will there be an integration where this function is not called? If there isn't I think it should not be part of the test, but included inside the initialization.
</blockquote></details>

1. Deciding the API might be premature. Moreover, I hope this mock will be deleted as soon as the real consensus is used.
2. Let us set the start height on setup - to simplify things as much as possible.

**Review by giladchase — COMMENTED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1296)* status: all files reviewed, 1 unresolved discussion (waiting on @ArniStarkware)

**Comment by ArniStarkware** [comment]
<details><summary><i>Previously, yair-starkware (Yair) wrote…</i></summary><blockquote>

IMO you remove the struct and leave just `run_consensus_for_end_to_end_test` that will get the batcher_client from outside
</blockquote></details>

Followed this attitude in #1352.

**Comment by ArniStarkware** [comment]
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

Followed this attitude in #1352.
</blockquote></details>

Which (theoretically) replaces this PR.

**Comment by yair-starkware** [comment]
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

Which (theoretically) replaces this PR.
</blockquote></details>

Looks better IMO
